openapi-cli/src/test/java/io/ballerina/openapi/generators/openapi/OpenApiConverterUtilsTest.java

```diff
@@ -174,9 +179,36 @@ public void testForNonHttpServices() {
         Assert.assertTrue(!Files.exists(tempDir.resolve("query_openapi.yaml")));
     }
 
+    @Test(description = "Given ballerina service has escape character")
+    public void testForRemovingEscapeIdentifier() throws IOException {
+        Path ballerinaFilePath = RES_DIR.resolve("escape_identifier.bal");
+        Path tempDir = Files.createTempDirectory("bal-to-openapi-test-out-" + System.nanoTime());
+        try {
+            String expectedYamlContent = getStringFromGivenBalFile(RES_DIR.resolve("expected_gen"),
+                    "escape_identifier.yaml");
+            OpenApiConverter openApiConverter = new OpenApiConverter();
+            openApiConverter.generateOAS3DefinitionsAllService(ballerinaFilePath, tempDir, null
+                    , false);
+            if (Files.exists(tempDir.resolve("v1_abc-hello_openapi.yaml"))) {
+                String generatedYaml = getStringFromGivenBalFile(tempDir, "v1_abc-hello_openapi.yaml");
+                generatedYaml = (generatedYaml.trim()).replaceAll("\\s+", "");
+                expectedYamlContent = (expectedYamlContent.trim()).replaceAll("\\s+", "");
+                Assert.assertTrue(generatedYaml.contains(expectedYamlContent));
+            } else {
+                Assert.fail("Yaml was not generated");
+            }
+        } catch (IOException e) {
+            Assert.fail("Error while generating the service. " + e.getMessage());
+        } finally {
+            deleteGeneratedFiles("v1_abc-hello_openapi.yaml", tempDir);
+            deleteDirectory(tempDir);
+            System.gc();
+        }
```

Comment:
yeah currently that function has some specific command name for the service base path, but given test, we need to change that service base path because of that  didn't used the same function here :)